mmocr/datasets/preparers/config_generator.py

```diff
@@ -0,0 +1,308 @@
+    """Base class for config generator.
+
+    Args:
+        data_root (str): The root path of the dataset.
+        task (str): The task of the dataset.
+        dataset_name (str): The name of the dataset.
+        train_anns (List[Dict], optional): A list of train annotation files
+            to appear in the base configs. Defaults to None.
+            Each element is typically a dict with the following fields:
+            - ann_file (str): The path to the annotation file relative to
+              data_root.
+            - dataset_postfix (str, optional): Affects the postfix of the
+              resulting variable in the generated config. If specified, the
+              dataset variable will be named in the form of
+              ``{dataset_name}_{dataset_postfix}_{task}_{split}``. Defaults to
+              None.
+        val_anns (List[Dict], optional): A list of val annotation files
+            to appear in the base configs, similar to ``train_anns``. Defaults
+            to None.
+        test_anns (List[Dict], optional): A list of test annotation files
+            to appear in the base configs, similar to ``train_anns``. Defaults
+            to None.
+        config_path (str): Path to the configs. Defaults to 'configs/'.
+    """
+
+    def __init__(
+        self,
+        data_root: str,
+        task: str,
+        dataset_name: str,
+        train_anns: Optional[List[Dict]] = None,
+        val_anns: Optional[List[Dict]] = None,
+        test_anns: Optional[List[Dict]] = None,
+        config_path: str = 'configs/',
+    ) -> None:
+        self.config_path = config_path
+        self.data_root = data_root
+        self.task = task
+        self.dataset_name = dataset_name
+        self._prepare_anns(train_anns, val_anns, test_anns)
+
+    def _prepare_anns(self, train_anns: Optional[List[Dict]],
+                      val_anns: Optional[List[Dict]],
+                      test_anns: Optional[List[Dict]]) -> None:
+        """Preprocess input arguments and stores these information into
+        ``self.anns``.
+
+        ``self.anns`` is a dict that maps the name of a dataset config variable
+        to a dict, which contains the following fields:
+        - ann_file (str): The path to the annotation file relative to
+          data_root.
+        - split (str): The split the annotation belongs to. Usually
+          it can be 'train', 'val' and 'test'.
+        - dataset_postfix (str, optional): Affects the postfix of the
+          resulting variable in the generated config. If specified, the
+          dataset variable will be named in the form of
+          ``{dataset_name}_{dataset_postfix}_{task}_{split}``. Defaults to
+          None.
+        """
+        self.anns = {}
+        for split, ann_list in zip(('train', 'val', 'test'),
+                                   (train_anns, val_anns, test_anns)):
+            if ann_list is None:
+                continue
+            if not isinstance(ann_list, list):
```

Comment:
isinstance(ann_list, (list, dict))
maybe better to support dict and list of dict?